### PR TITLE
IGNITE-21718 Extract AbstractPageMemoryMvPartitionStorage volatile state into a class

### PR DESCRIPTION
**Pull request by sashapolo**
https://issues.apache.org/jira/browse/IGNITE-21718

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

**Review by rpuch — COMMENTED** [review]
I like the idea. Let's also hear @tkalkirill

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/PartitionStorageMutableState.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+package org.apache.ignite.internal.storage.pagememory.mv;
+
+import org.apache.ignite.internal.storage.pagememory.index.freelist.IndexColumnsFreeList;
+import org.apache.ignite.internal.storage.pagememory.index.meta.IndexMetaTree;
+import org.apache.ignite.internal.storage.pagememory.mv.gc.GcQueue;
+
+/**
+ * Simple holder class that comprises the volatile state of {@link AbstractPageMemoryMvPartitionStorage}.
+ */
+public class PartitionStorageMutableState {
```
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/PartitionStorageMutableState.java:27]: It's not that straightforward, unfortunately. This class is used in a `protected` field and if I make this class package private, I get a warning from IDEA that the class is exposed outside of its defined scope.

[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/PartitionStorageMutableState.java]: Warning IDEAS does this mean the end of the world is approaching?

[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/PartitionStorageMutableState.java]: Fixed

**Review by tkalkirill — APPROVED** [review]

**Pull request merged**